src/ripple/protocol/impl/STValidation.cpp

```diff
@@ -90,6 +92,12 @@ STValidation::getConsensusHash() const
     return getFieldH256(sfConsensusHash);
 }
 
+std::uint32_t
+STValidation::getNetworkID() const
+{
+    return getFieldU32(sfNetworkID);
```

Comment:
I think you should return optional here. Something like:

```
std::optional<std::uint32_t>
STValidation::getNetworkID() const
{
    if (!isFieldPresent(sfNetworkID))
        return std::nullopt;
    return getFieldU32(sfNetworkID);
}
```

There may be a way to do this more efficiently using the weird `~sfNetworkID` syntax but I'm still learning code.